ui/mods/msu/popup.js

```diff
@@ -96,25 +100,62 @@ MSUPopup.prototype.showRawText = function (_data)
 {
 	if (_data.forceQuit)
 	{
+		this.mTitle.text("Fatal Mod Error");
 		this.mFooterContainer.find(".ok-button:first").trigger('force-quit')
 	}
-	if (this.isVisible())
+	else
 	{
-		this.mTextContainer.html(this.mTextContainer.html() + "<br>" + _data.text);
+		this.mTitle.text("Mod Error");
 	}
-	else
+	this.mListScrollContainer.append($('<div class="mod-raw-text">' + _data.text + '</div>'));
+	if (!this.isVisible())
 	{
-		this.mTextContainer.html(_data.text);
 		this.show();
 	}
 }
 
-MSUPopup.prototype.hide = function ()
+MSUPopup.prototype.showModUpdates = function (_mods)
 {
-	this.mTextContainer.html("");
+	this.mTitle.text("Mod Updates Available");
 	var self = this;
+	$.each(_mods, function (_key, _modInfo)
+	{
+		var modInfoContainer = $('<div class="msu-mod-info-container"/>');
+		self.mListScrollContainer.append(modInfoContainer)
+		modInfoContainer.append($('<div class="mod-name title title-font-big font-bold font-color-title">' + _modInfo.name + '</div>'));
+		var githubContainer = $('<div class="l-github-button"/>');
+		modInfoContainer.append(githubContainer);
+		var githubButton = githubContainer.createImageButton(Path.GFX + "mods/msu/logos/github-32.png", function ()
+		{
+			openURL(_modInfo.githubURL);
+		});
+		if (_modInfo.githubURL == null) githubButton.attr('disabled', true);
```

Comment:
It's black and white so this is probably not very visible. Just hide them if they're inactive?

Reply:
That's fair ig, though it'd never come up rn

Reply:
Yeah ik but for completion's sake